source/common/http/utility.cc

```diff
@@ -636,18 +636,34 @@ bool Utility::sanitizeConnectionHeader(Http::RequestHeaderMap& headers) {
 const std::string& Utility::getProtocolString(const Protocol protocol) {
   switch (protocol) {
   case Protocol::Http10:
-    return Headers::get().ProtocolStrings.Http10String;
+    return StreamInfo::ProtocolStrings::get().Http10String;
   case Protocol::Http11:
-    return Headers::get().ProtocolStrings.Http11String;
+    return StreamInfo::ProtocolStrings::get().Http11String;
   case Protocol::Http2:
-    return Headers::get().ProtocolStrings.Http2String;
+    return StreamInfo::ProtocolStrings::get().Http2String;
   case Protocol::Http3:
-    return Headers::get().ProtocolStrings.Http3String;
+    return StreamInfo::ProtocolStrings::get().Http3String;
   }
 
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+absl::optional<const Protocol> Utility::getProtocol(absl::Span<const std::string> protocols) {
+  for (auto it = protocols.rbegin(); it != protocols.rend(); it++) {
```

Comment:
I assume you are going in reverse for a reason? Comment?

Reply:
Sure, sorry if it was not clear but it's basically append-only history of discovered protocols, so looking from the back is right.